examples/pytorch/graphsaint/README.md

```diff
@@ -26,74 +26,120 @@ All datasets used are provided by Author's [code](https://github.com/GraphSAINT/
-| --- | --- | --- |
-| Sampling(Running) | 0.72 | 0.56 |
-| Sampling(DGL) | 0.50 | 0.72 |
-| Normalization(Running) | 0.68 | 2.62 |
-| Normalization(DGL) | 0.61 | 0.38 |
+| Method      | PPI  | Flickr | Reddit | Yelp | Amazon |
+| --- | --- | --- | --- | --- | --- |
+| Running | 1.4 | 3.18 | 13.88 | 39.02 | 164.35 |
+| DGL_offline | 3.08 | 1.91 | 53.59 | 48.94 | 1469.52 |
+| DGL_online | 3.04 | 1.87 | 52.01 | 48.38 | 1432.79 |
 
 #### Random walk sampler
 
-| Method | PPI | Flickr |
-| --- | --- | --- |
-| Sampling(Running) | 0.83 | 1.22 |
-| Sampling(DGL) | 0.28 | 0.63 |
-| Normalization(Running) | 0.87 | 2.60 |
-| Normalization(DGL) | 0.70 | 0.42 |
+| Method      | PPI  | Flickr | Reddit | Yelp | Amazon |
+| --- | --- | --- | --- | --- | --- |
+| Running | 1.7 | 3.82 | 16.97 | 43.25 | 355.68 |
+| DGL_offline | 3.10 | 2.18 | 10.57 | 21.52 | 153.93 |
+| DGL_online | 3.05 | 2.13 | 11.01 | 22.23 | 151.84 |
+
+## Test std of sampling time
+
+- We've run experiments 10 times repeatedly to test average and standard deviation of sampling time. Here we just test sampling time without model performance. Moreover, for efficient testing, the hardware and config employed here are not the same as the experiments above, so the sampling time might be a bit different from that above. But we keep the environment consistent in all experiments below.
+
+> The value is (average, std)
+
+### Random node sampler
+
+| Method                    | PPI             | Flickr       | Reddit        | Yelp          | Amazon          |
+| ------------------------- | --------------- | ------------ | ------------- | ------------- | --------------- |
+| DGL_Sampling(std)         | 2.618, 0.004    | 3.017, 0.507 | 35.356, 2.363 | 69.913, 6.3   | 888.025, 16.004 |
+| DGL_Normalization(std)    | Small to ignore | 0.008, 0.004 | 0.26, 0.047   | 0.189, 0.0288 | 2.443, 0.124    |
+|                           |                 |              |               |               |                 |
+| author_Sampling(std)      | 0.788, 0.661    | 0.728, 0.367 | 8.931, 3.155  | 27.818, 1.384 | 295.597, 4.928  |
+| author_Normalization(std) | 0.665, 0.565    | 4.981, 2.952 | 17.231, 7.116 | 47.449, 2.794 | 279.241, 17.615 |
+
+### Random edge sampler
+
+| Method                    | PPI             | Flickr       | Reddit        | Yelp          | Amazon |
+| ------------------------- | --------------- | ------------ | ------------- | ------------- | ------ |
+| DGL_Sampling(std)         | 3.554, 0.292    | 4.722, 0.245 | 47.09, 2.76   | 75.219, 6.442 |        |
+| DGL_Normalization(std)    | Small to ignore | 0.005, 0.007 | 0.235, 0.026  | 0.193, 0.021  |        |
+|                           |                 |              |               |               |        |
+| author_Sampling(std)      | 0.802, 0.667    | 0.761, 0.387 | 6.058, 2.166  | 13.914, 1.864 |        |
+| author_Normalization(std) | 0.667, 0.570    | 5.180, 3.006 | 15.803, 5.867 | 44.278, 5.853 |        |
```

Comment:
Have you tried sampling without replacement for edge sampler on Amazon? If that works, we can have numbers as well.

Reply:
I can take a try later.